waspc/data/Generator/templates/sdk/wasp/client/operations/hooks.ts

```diff
@@ -259,10 +263,10 @@ function makeRqOptimisticUpdateOptions<ActionInput, CachedData>(
     );
 
     // We're using a Map to correctly serialize query keys that contain objects.
-    const previousData = new Map();
+    const previousData: Map<QueryKey, CachedData | undefined> = new Map();
```

Comment:
Tl;dr everything about this part of the code is covered in https://github.com/wasp-lang/wasp/issues/2017#issue-2283681572, but it turns out this was actually correct (but would be nicer if we handled it more explicitly).